metadata/metadata.c

```diff
@@ -148,14 +151,29 @@ Allocator *create_allocator(VirtualDisk__ErasureCodeProfile ec_profile)
     return result;
 }
 
+static void set_allocator(Allocator *allocator) {
+    metadata_disk_allocator = allocator;
+}
+
 Metadata *new_metadata()
 {
     Metadata *metadata = (Metadata *)malloc(sizeof(Metadata));
     metadata__init(metadata);
+    set_allocator(create_allocator(VIRTUAL_DISK__ERASURE_CODE_PROFILE__EC_4_2P));
```

Comment:
I think we should store the system wide EC profile in the metadata proto so it gets persisted and is configured system wide.

```suggestion
Metadata *new_metadata(VirtualDisk__ErasureCodeProfile ec_profile)
{
    Metadata *metadata = (Metadata *)malloc(sizeof(Metadata));
    metadata__init(metadata);
    metadata->ec_profile = ec_profile;
    set_allocator(create_allocator(ec_profile));
```

Reply:
I made it global and configurable once - upon metadata creation.